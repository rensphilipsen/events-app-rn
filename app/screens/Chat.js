// @flow
import React, { PureComponent } from 'react';
import { Bubble, GiftedChat } from 'react-native-gifted-chat';
import Fire from '../utils/Fire';
import connect from 'react-redux/es/connect/connect';
import { COLOR } from '../styles/theme';

class Chat extends PureComponent {

    /**
     * Local state
     *
     * @type {{messages: Array}}
     */
    state = {
        messages: [],
    };

    /**
     * Component did mount
     */
    componentDidMount() {
        new Fire(this.props.user.email).on(message =>
            this.setState(previousState => ({
                messages: GiftedChat.append(previousState.messages, message),
            }))
        );
    }

    /**
     * Component should unmount
     */
    componentWillUnmount() {
        new Fire(this.props.user.email).off();
    }

    /**
     * Fetch user from redux store
     *
     * @returns {{name: *, _id: string | string}}
     */
    get user() {
        return {
            name: this.props.user.name,
            _id: this.props.user.email,
        };
    }

    renderBubble = (props) => {
        return (
            <Bubble
                {...props}
                wrapperStyle={{
                    right: {
                        backgroundColor: COLOR.PRIMARY
                    },
                }}
            />
        );
    };

    /**
     * Render method.
     *
     * @returns {*}
     */
    render() {
        return (
            <GiftedChat
                messages={this.state.messages}
                onSend={new Fire(this.props.user.email).send}
                renderBubble={this.renderBubble}
                locale={'nl'}
                user={this.user}
            />
        );
    }
}

/**
 * All the VALUES from the Redux store that should be available within the props of this component
 *
 * @param state
 * @returns {{user: *}}
 */
const mapStateToProps = state => {
    return {
        user: state.users
    };
};

/**
 * All the METHODS from the Redux store that should be available within the props of this component
 */
const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(Chat)