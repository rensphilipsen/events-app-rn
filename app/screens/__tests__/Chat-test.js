// Copyright (c) 2014-present, Facebook, Inc. All rights reserved.

/**
 * Sample React Native Snapshot Test
 */

'use strict';

import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import ShallowRenderer from 'react-test-renderer/shallow';
import Chat from "../Chat";

jest.setTimeout(15000);

it('renders correctly', () => {
	const renderer = new ShallowRenderer();
	renderer.render(<Chat/>);
	const tree = renderer.getRenderOutput();
	expect(tree).toMatchSnapshot();
});