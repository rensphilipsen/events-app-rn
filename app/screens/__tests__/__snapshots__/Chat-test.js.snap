// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<GiftedChat
  bottomOffset={0}
  dateFormat="ll"
  forceGetKeyboardHeight={false}
  imageProps={Object {}}
  inverted={true}
  isAnimated={true}
  isLoadingEarlier={false}
  keyboardShouldPersistTaps="never"
  lightboxProps={Object {}}
  listViewProps={Object {}}
  loadEarlier={false}
  locale={null}
  maxInputLength={null}
  messageIdGenerator={[Function]}
  messages={
    Array [
      Object {
        "_id": 1,
        "createdAt": "2018-12-12 00:00",
        "text": "Hello",
        "user": Object {
          "_id": 2,
          "avatar": "https://placeimg.com/140/140/any",
          "name": "React Native",
        },
      },
    ]
  }
  minInputToolbarHeight={44}
  onInputTextChanged={null}
  onLoadEarlier={[Function]}
  onLongPress={null}
  onPressActionButton={null}
  onPressAvatar={null}
  onSend={[Function]}
  placeholder="Type a message..."
  renderAccessory={null}
  renderActions={null}
  renderAvatarOnTop={false}
  renderBubble={null}
  renderChatFooter={null}
  renderComposer={null}
  renderCustomView={null}
  renderDay={null}
  renderFooter={null}
  renderInputToolbar={null}
  renderLoadEarlier={null}
  renderLoading={null}
  renderMessage={null}
  renderMessageImage={null}
  renderMessageText={null}
  renderSend={null}
  renderSystemMessage={null}
  renderTime={null}
  showUserAvatar={false}
  textInputProps={Object {}}
  timeFormat="LT"
  user={
    Object {
      "_id": 1,
    }
  }
/>
`;
